chain-core/src/property.rs

```diff
@@ -222,6 +241,23 @@ pub trait LeaderSelection {
     fn is_leader_at(&self, date: <Self::Block as Block>::Date) -> Result<bool, Self::Error>;
 }
 
+/// the settings of the blockchain this is something that can be used to maintain
+/// the blockchain protocol update details:
+///
+pub trait Settings {
+    type Update: Update;
+    type Block: Block;
+    type Error: std::error::Error;
+
+    /// read the block update settings and see if we need to store
+    /// updates. Protocols may propose vote mechanism, this Update
+    /// and the settings need to keep track of these here.
+    fn diff(&self, input: &Self::Block) -> Result<Self::Update, Self::Error>;
+
+    /// apply the Update to the LeaderSelection
```

Comment:
This mentions `LeaderSelection` while the trait is `Settings`. I think this needs to be fixed or further clarified how leader selection figures here.